src/modules/Exchange.ts

```
@@ -109,6 +116,85 @@ export class Exchange {
     );
   }
 
+  public async proxyDepositWithZeroEx(
+    {
+      humanSellAmount,
+      sellTokenAddress,
+      starkKey,
+      positionId,
+      slippagePercentage = '1.0',
+    }: {
+      humanSellAmount: string,
+      sellTokenAddress: string,
+      starkKey: string,
+      positionId: BigNumberable,
+      slippagePercentage?: string,
+    },
+    options?: SendOptions,
+  ): Promise<TxResult> {
+    validateSlippage(slippagePercentage);
+
+    const sellAmount: string = humanCollateralAmountToUint256(humanSellAmount);
+
+    const zeroExRequest = await getZeroExSwapQuote(
+      {
+        sellAmount,
+        sellTokenAddress,
+        buyTokenAddress: USDC_ADDRESSES[this.contracts.networkId],
```

Should make sure we throw if this is undefined